x-pack/plugins/fleet/server/services/preconfiguration.test.ts

```diff
@@ -217,392 +291,527 @@ describe('policy preconfiguration', () => {
+      const soClient = getPutPreconfiguredPackagesMock();
+      const esClient = elasticsearchServiceMock.createClusterClient().asInternalUser;
+      const policy: PreconfiguredAgentPolicy = {
+        name: 'Test policy',
+        namespace: 'default',
+        id: 'test-id',
+        package_policies: [],
+        is_managed: true,
+      };
+      mockConfiguredPolicies.set('test-id', policy);
+
+      const { policies, nonFatalErrors: nonFatalErrorsB } =
+        await ensurePreconfiguredPackagesAndPolicies(
+          soClient,
+          esClient,
+          [policy],
+          [],
+          mockDefaultOutput,
+          DEFAULT_SPACE_ID
+        );
+      expect(spyAgentPolicyServiceUpdate).not.toBeCalled();
+      expect(policies.length).toEqual(1);
+      expect(policies[0].id).toBe('test-id');
+      expect(nonFatalErrorsB.length).toBe(0);
+    });
   });
 
-  it('should not update a managed policy if a top level field has not changed', async () => {
-    const soClient = getPutPreconfiguredPackagesMock();
-    const esClient = elasticsearchServiceMock.createClusterClient().asInternalUser;
-    const policy: PreconfiguredAgentPolicy = {
-      name: 'Test policy',
-      namespace: 'default',
-      id: 'test-id',
-      package_policies: [],
-      is_managed: true,
-    };
-    mockConfiguredPolicies.set('test-id', policy);
+  describe('with bundled packages', () => {
+    beforeEach(() => {
+      mockInstalledPackages.clear();
+      mockedGetBundledPackages.mockReset();
+    });
 
-    const { policies, nonFatalErrors: nonFatalErrorsB } =
-      await ensurePreconfiguredPackagesAndPolicies(
+    it('should install each bundled package', async () => {
+      mockedGetBundledPackages.mockResolvedValue([
+        {
+          name: 'test_package',
+          buffer: Buffer.from('test_package'),
+        },
+
+        {
+          name: 'test_package_2',
+          buffer: Buffer.from('test_package_2'),
+        },
+      ]);
+
+      const soClient = getPutPreconfiguredPackagesMock();
+      const esClient = elasticsearchServiceMock.createClusterClient().asInternalUser;
+
+      const { policies, packages, nonFatalErrors } = await ensurePreconfiguredPackagesAndPolicies(
         soClient,
         esClient,
-        [policy],
         [],
+        [
+          {
+            name: 'test_package',
+            version: 'latest',
+          },
+          {
+            name: 'test_package_2',
+            version: 'latest',
+          },
+        ],
         mockDefaultOutput,
         DEFAULT_SPACE_ID
       );
-    expect(spyAgentPolicyServiceUpdate).not.toBeCalled();
-    expect(policies.length).toEqual(1);
-    expect(policies[0].id).toBe('test-id');
-    expect(nonFatalErrorsB.length).toBe(0);
+
+      expect(policies).toEqual([]);
+      expect(packages).toEqual(['test_package-1.0.0', 'test_package_2-1.0.0']);
+      expect(nonFatalErrors).toEqual([]);
+    });
+
+    describe('package updates', () => {
```

Comment:
@joshdover I added some tests here that should cover the "update package from disk" cases you mentioned previously.